init.d/02.setup-resources/manifests/frontend/game.js

```diff
@@ -56,10 +55,41 @@ function hideEndscreen() {
     document.getElementById("endscreen").style.display = "none";
 }
 
+function finish() {
+    if(isNewGame()) {
+        localStorage.removeItem("activeGame");
+        location.reload();
+    } else {
+        flashResetInstruction();
+    }
+}
+
+function flashResetInstruction() {
+    const resetInstruction = document.getElementById('resetInstruction');
+    const originalBackground = resetInstruction.style.backgroundColor;
+    resetInstruction.style.backgroundColor = "rgba(255, 0, 0, 0.3)";
+    setTimeout(() => {
+        resetInstruction.style.backgroundColor = originalBackground;
+    }, 1000);
+}
+
 function showEndscreen() {
     document.getElementById("endscreen").style.display = "inline";
 }
 
+function isNewGame() {
+    checkGameState().then(() => {
+        return Object.values(gameState.puzzles)
+            .map((puzzle) => puzzle.solved)
+            .every((solved) => solved === false);
+    });
+}
```

Comment:
Right now this doesn't do anything (since the function itself doesn't return anything), this is another case where Typescript would be a nice help to show that the function is of type `() => void`.

You would at least have to `return checkGameState()...` (or ideally `return await checkGameState()`.

I think in this case it might be enough to do:
```suggestion
function isNewGame() {
    return Object.values(gameState.puzzles)
        .map((puzzle) => puzzle.solved)
        .every((solved) => solved === false);
}
```
Since `checkGameState()` is already called by the game loop every 3 seconds

Reply:
Left the checkGameState so there is no wait time between cluster restart and game restart.
Changed this to async and put the checkGameState in front, so the boolean is returned.

Reply:
And yes, please let's bring up typescript in the next meeting!